### ISSUE TITLE
Snippets

### ISSUE THREAD

**Issue by exerro** [ebernerd/WaveLite#13]
`editor.snippet(cursor, snippet)` will write the snippet. Snippets are like normal code, except you can put things like `${1:name}` (which will show just `name`) but by pressing tab it will cycle through `1 .. n`, selecting the text (`name` in that case), allowing you to modify it with ease. For example...

``` lua
editor.map( editor.snippet, nil, [[
${4:local }function ${1:FunctionName}( ${2:...} )
\t${3: -- code}
end]] )
```

These should work well even with multiple clients, so some kind of tag should be assigned to the cursor when writing the snippet.

**Comment by demhydraz** [comment]
Should be a plugin, IMO.

**Comment by exerro** [comment]
Yeah, I guess that makes sense. Need to work on a (better?) way of allowing plugins to communicate then.
